build/xtask/src/config.rs

```diff
@@ -551,13 +551,14 @@ impl BuildConfig<'_> {
             None => PathBuf::from("cargo"),
         });
 
-        // nightly features that we use: asm_sym,named-profiles,naked_functions
-        // cmse_nonsecure_entry,array_methods,destructuring_assignment
+        // nightly features that we use: asm_sym,asm_const,
+        // named-profiles,naked_functions,cmse_nonsecure_entry,array_methods,
+        // destructuring_assignment
         // nightly features that our dependencies use: proc_macro_span,backtrace
         // asm,llvm_asm
 
         cmd.arg(
-            "-Zallow-features=asm,asm_sym,named-profiles,naked_functions,\
+            "-Zallow-features=asm_sym,asm_const,named-profiles,naked_functions,\
```

Comment:
Should be able to remove `destructuring_assignment` right?

Reply:
Yes!

I probably should just re-do the list again manually... I guess I really don't want just an allowlist, but like, bidirectional checking that they're actually required, ha!

I'll fix this soon

Reply:
I've re-done this now; our dependencies no longer use `asm` and `llvm_asm` too :)